_pages/algorithm-development/source-control-management.md

```diff
@@ -428,3 +434,62 @@ Yes! Upon any change to your repository's name or owner we receive a webhook whi
+
+If you're new to Git, we recommend [this tutorial series](https://try.github.io/) by GitHub.
+
+**Web IDE Support** At this time we do not support editing source code in our web app for Bitbucket Server-hosted algorithms. 
+{: .notice-info}
+
+
+#### Creating Algorithms with Bitbucket Server
+
+To host your algorithm on Bitbucket Server, simply select the Bitbucket Server instance you wish to use when creating your algorithm.
+
+Once you've selected your Bitbucket Server instance, you will be able to host algorithm source code in Bitbucket Server.
+
+You can customize two aspects of the Bitbucket Server repositories that are created for your algorithms: the repository's owner, and its name.
+
+By default, all Bitbucket Server respositories are the project configured by your administrator. These proojects are listed in the "Repository Owner" dropdown:
+
+You may also customize the name of the repository that is created. By default the name of your algorithm is used. 
+
+If you decide to use an alternative name, we recommend that you use only letters, numbers, hyphens, and/or underscores. While Bitbucket Server will accept characters outside this range, they will replace any unsupported characters with an underscore, resulting in an undesirable repository name.
+
+When we create your repository, we associate the following with it:
+
+- **A Personal Access Token**: Bitbucket Server personal access tokens allow appropriate access to specific repositories, and are Bitbucket Server’s prescribed means by which external services can fetch code for building and deploying. These tokens are tied to individual permissions, and as such will not allow Algorithmia to continue building an algorithm if the permissions of the Personal Access Token changes. We also use the Personal Access Token to obtain your repository's commit log (to display changes when publishing versions) and README.md (for use as algorithm documentation).
+- **Webhooks**: We set up webhooks to receive notifications about changes to your repo, such as when there's a change to its default branch.
+
+#### Updating Bitbucket Server-Hosted Algorithms
+
+Once you've created your algorithm, any commits to your Bitbucket Server repository's _default_ branch will result in a build on Algorithmia. At creation, your repository's default branch will be your `master` branch, but you can change this at any time in your repository's settings.
+
+You can view your algorithm's builds by heading to its landing page and clicking the "Builds" tab if you are the algorithm's owner.
+
+#### Publishing Bitbucket Server-Hosted Algorithms
+
+When you're ready to create a new version of your algorithm, visit that algorithm's page and select the __Builds__ tab. Locate the build corresponding to specific commit SHA you wish to use, and then click __Publish__ on the right side of the row.
```

Comment:
Formatting of "Builds" from previous paragraph and __Builds__ from this paragraph is different.